middleman_protocol/middleman_protocol/constants.py

```diff
@@ -0,0 +1,16 @@
+from enum import IntEnum
+from enum import unique
+
+
+@unique
+class PayloadType(IntEnum):
+    GOLEM_MESSAGE               = 0  # A serialized Golem message.
+    ERROR                       = 1  # An error code and an error message.
+    AUTHENTICATION_CHALLENGE    = 2  # A random string of bytes.
+    AUTHENTICATION_RESPONSE     = 3  # A digital signature of the content sent as authentication challenge.
+
+
+# Random sequence of bytes used as separator between messages. Should be placed after frame.
+FRAME_SEPARATOR = b'\\x1d\\xb7{\\xb0\\xb9\\x16\\xc8f/\\xd7\\xf0\\xc7\\x06\\x07\\x1e\\xa2'
+
+FRAME_SEPARATOR_LENGTH = len(FRAME_SEPARATOR)
```

Comment:
I wouldn't even worry about it being recalculated. It's such an insignificant overhead compared to other operations. I'm only concerned with readability here.

And aren't Python strings stored internally with length anyway?

Reply:
From what I have read they are stored with length equal to (size + 1), where that `+ 1` is for the finishing byte.

How about this - please check in which places it is used, and if replacing it with `len(FRAME_SEPARATOR)` will increase readability. For example there is a line

```
received_data[:-FRAME_SEPARATOR_LENGTH] == FRAME_SEPARATOR
```

where I think it fits perfectly. If after checking you still consider it will increase readability, I will change it.